Automattic/woocommerce-payments
# refactor: ECE utility for button UI interactions

Fixes #

#### Changes proposed in this Pull Request









In the "Tokenized Paymenr Request" implementation, we had a "Button UI" class, which allows to interact with the jQuery elements around the button and slimmed a bit the wcPayEce class from some methods: https://github.com/Automattic/woocommerce-payments/blob/0df81cc8f4ddf157dcb2a12c72a410e80ed2b9c3/client/tokenized-payment-request/button-ui.js

I'd like to bring it to the ECE implementation, so it can also be reflected in the Tokenized ECE implementation.

I didn't isolate these changes to the Tokenized ECE implementation for a couple of reasons:
- They seem on the safe side
- I'd like to keep the differences between the main ECE and the tokenized ECE at a minimum

I'll also copy the same changes to the tokenized ECE ( https://github.com/Automattic/woocommerce-payments/tree/develop/client/tokenized-express-checkout ), once this is merged.

Since this is just moving the methods in an isolated class, the same test coverage should be present as before.

#### Testing instructions





The changes don't affect block-based elements (like block-based cart/checkout).
These changes affect product page, shortcode cart & checkout, pay-for-order.

- Ensure you have the "Enable Cart-Token implementation for ECE" feature flag disabled in the Dev Tools
- As a merchant, ensure you have GooglePay/ApplePay enabled in the WooPayments settings (for product pages, cart, checkout)
- As a customer, go to a product page for a simple product
- You should see the GooglePay button in Chrome
- Interact with the product quantities, the button should update
- You can click on the button and you should be able to interact with it
- Close the dialog, add the product to the cart
- Go to the shortcode cart page, you should be able to see the interact with the GooglePay button
- Go to the shortcode checkout page, you should be able to see the interact with the GooglePay button

-------------------


**Post merge**
